### ISSUE TITLE
When I can't access a pod, tell me to tell the person who game the the link to give me the sharable binder link instead

### ISSUE THREAD

**Issue by das-g** [jupyterhub/binderhub#457]
(originally suggested [on Twitter](https://twitter.com/betatim/status/964627934165323776) by @betatim )

When someone [has published a personal pod URL<sup>1</sup>](https://twitter.com/pythonsummit/status/964442816637296640) (or has shared it with me, see #160), I won't (and probably shouldn't) be able to access that pod.

Chances are, that they didn't want to give me access to their personal pod, but instead to a new pod based on the same publicly hosted notebook file or repo, which they could have achieved by sharing the ["binder link"<sup>2</sup>](https://mybinder.readthedocs.io/en/latest/introduction.html#generating-a-sharable-binder-link).

Thus, the error page I land on when following the pod URL should explain the difference between these two (if possible, without requiring prior knowledge of kubernetes lingo or other implementation details) **tell me to tell the person who gave me the link how to fix it** (i.e., to share the corresponding sharable binder link instead).

(Could be combined with #458, but either suggestion could be implemented without the other, so I'm filing them as separate issues.)

---
<sup>1</sup>On the mybinder service, an URL beginning with <code>https://<strong>hub.</strong>mybinder.org</code> &hellip;
<sup>2</sup>On the mybinder service, the URL of form `https://mybinder.org/v2/<provider-name>/<org-name>/<repo-name>/<branch|commit|tag>?filepath=<path/to/notebook.ipynb>`

**Comment by minrk** [comment]
Indeed, both this and #458 are closely related, and mostly involve custom templates for pages in two cases:

1. the login.html page for the case where a pod is still running and you don't have access to it
2. the 403 page served by the Hub when no such Binder is running at all

The login.html case isn't too hard, and can be addressed via the repo2docker appendix, as mentioned in #458. The other case is a little trickier, but should also be doable.

**Comment by minrk** [comment]
The 403 case is less solvable (as discussed in #459) because that event occurs when everything's been cleaned up, and the Hub only has access to the now-defunct URL. The URL is not itself enough information to construct the original binder link. We can use a custom template to improve that page so that it says "This was a link to a running binder, tell whoever gave you this link to give you the right one", but we can't do better than that without recording the hub_user:binder_url mapping in some data store that survives after a pod (and user) has been culled.

A custom Hub template does mean a custom Hub image, unless we build custom hub templates into the Hub chart itself. Perhaps that's what we should do.
